docs/book/src/topics/openstack-cluster-identity.md

```diff
@@ -0,0 +1,100 @@
+# Centralized credentials with OpenStackClusterIdentity
+
+This guide explains how to centralize OpenStack credentials using a cluster-scoped OpenStackClusterIdentity and reference it from clusters and machines.
+
+## Overview
+- OpenStackClusterIdentity (cluster-scoped): stores a reference to a Secret that contains `clouds.yaml` (and optional `cacert`), and optionally restricts which namespaces may use it via `namespaceSelector`.
+- OpenStackIdentityReference (on OpenStackCluster/OpenStackMachine/OpenStackServer): carries `type`, `name`, and `cloudName`.
+  - `type: Secret` (default): `name` is the Secret name in the same namespace.
+  - `type: ClusterIdentity`: `name` is the OpenStackClusterIdentity name; Secret location is taken from the identity.
+  - For both types, `cloudName` is required and selects the entry in `clouds.yaml`.
+
+## Prerequisites
+- A Secret containing OpenStack credentials in `clouds.yaml`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openstack-credentials
+  namespace: capo-system
+stringData:
+  clouds.yaml: |
+    clouds:
+      openstack:
+        auth:
+          auth_url: https://keystone.example.com/
+          application_credential_id: <id>
+          application_credential_secret: <secret>
+        region_name: RegionOne
+        interface: public
+        identity_api_version: 3
+        auth_type: v3applicationcredential
+  # Optional CA certificate
+  # cacert: |
+  #   -----BEGIN CERTIFICATE-----
+  #   ...
+  #   -----END CERTIFICATE-----
+```
+
+## Create an OpenStackClusterIdentity
+- Optionally restrict which namespaces can use it with `namespaceSelector`.
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+kind: OpenStackClusterIdentity
+metadata:
+  name: production-openstack
+spec:
+  secretRef:
+    name: openstack-credentials
+    namespace: capo-system
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: [team-a, team-b]
+```
+
+## Reference the identity from OpenStackCluster
+- Use `type: ClusterIdentity`, specify the identity `name`, and the `cloudName` to select the clouds.yaml entry.
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: OpenStackCluster
+metadata:
+  name: cluster-a
+  namespace: team-a
+spec:
+  identityRef:
+    type: ClusterIdentity
```

Comment:
I think it might be clearer to rename `type` to `kind`, and expect exactly `kind: OpenStackClusterIdentity` as a value, aligning with the similar AWS implementation. What do you think?

Reply:
is it about the terminology? I chose type over kind due to the history in CAPO.

From CAPO: 
> Incidentally, an older version of the CAPO API actually had a type field, but we removed it because we only had a single type. Perhaps that was short-sighted, but it's easy enough to add back.

ref: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2628#issuecomment-3132112472 


About ClusterIdentity vs OpenStackClusterIdentity, the code does use OpenStackClusterIdentity for the CRD. This is shortened only in the identityRef for brevity. Especially since the two choices are secret and clusterIdentity, thought it's best to keep it like this. Let me know what you think.

Reply:
Got it, thanks. If this proposal has already been approved by the CAPO community, then we should align with it. Let's keep it as is then.